client/src/Forms/FormForm.js

```diff
@@ -1,4 +1,153 @@
+import { useEffect, useState } from 'react';
+import { useHistory, useParams } from 'react-router-dom';
+import { StatusCodes } from 'http-status-codes';
+import classNames from 'classnames';
+
+import Api from '../Api';
+import UnexpectedError from '../UnexpectedError';
+import ValidationError from '../ValidationError';
+
 function FormForm() {
-  return <></>;
+  const { id } = useParams();
+  const history = useHistory();
+  const [error, setError] = useState(null);
+  const [form, setForm] = useState({
+    name: '',
+    logo: '',
+    year: '',
+    school: '',
+    url: '',
+    lang: '',
+  });
+  const [isUploading, setUploading] = useState(false);
+
+  useEffect(
+    function () {
+      if (id) {
+        Api.form.get(id).then((response) => setForm(response.data));
+      }
+    },
+    [id]
+  );
+
+  function onChange(event) {
+    const newForm = { ...form };
+    newForm[event.target.name] = event.target.value;
+    setForm(newForm);
+  }
+
+  async function onSubmit(event) {
+    event.preventDefault();
+    setError(null);
+    try {
+      if (id) {
+        await Api.form.update(id, form);
+      } else {
+        await Api.form.create(form);
+      }
+      history.push('/form');
+    } catch (error) {
+      if (error.response?.status === StatusCodes.UNPROCESSABLE_ENTITY) {
+        setError(new ValidationError(error.response.data));
+      } else {
+        setError(new UnexpectedError());
+      }
+    }
+  }
+
+  return (
+    <main className="container">
+      <h1>Form</h1>
+      <form onSubmit={onSubmit}>
+        <div className="mb-3">
+          <label htmlFor="name">Name: </label>
+          <input
+            className={classNames('form-control', { 'is-invalid': error?.errorsFor?.('name') })}
+            type="text"
+            id="name"
+            name="name"
+            required
+            onChange={onChange}
+            value={form.name}
+          />
+          {error?.errorMessagesHTMLFor?.('name')}
+        </div>
+
+        <div className="mb-3">
+          <label htmlFor="logo">Logo: </label>
+          <input
+            className={classNames('form-control', { 'is-invalid': error?.errorsFor?.('logo') })}
+            type="text"
+            id="logo"
+            name="logo"
+            required
+            onChange={onChange}
+            value={form.logo}
+          />
+          {error?.errorMessagesHTMLFor?.('logo')}
+        </div>
```

Comment:
For the `logo` attribute, let's use the PhotoUploader component instead of a textbox so that an actual image file can be attached... take a look at ResourceForm.js for an example of how it is used there...